File: sdk/dotnet/Pulumi/Resources/DeleteBehaviour.cs
// Copyright 2016-2021, Pulumi Corporation

using System.Collections.Generic;

namespace Pulumi
{
    public enum DeleteBehaviour {
        Delete = Pulumirpc.DeleteBehaviour.Delete,
        Drop = Pulumirpc.DeleteBehaviour.Drop,
        Protect = Pulumirpc.DeleteBehaviour.Protect,
    }
}

File: sdk/go/common/resource/delete_behaviour.go
package resource

import pulumirpc "github.com/pulumi/pulumi/sdk/v3/proto/go"

type DeleteBehaviour int32

const (
	DeleteBehaviourDelete  DeleteBehaviour = DeleteBehaviour(pulumirpc.RegisterResourceRequest_DELETE)
	DeleteBehaviourDrop                    = DeleteBehaviour(pulumirpc.RegisterResourceRequest_DROP)
	DeleteBehaviourProtect                 = DeleteBehaviour(pulumirpc.RegisterResourceRequest_PROTECT)
)

func (s DeleteBehaviour) String() string {
	switch s {
	case DeleteBehaviourDelete:
		return "delete"
	case DeleteBehaviourDrop:
		return "drop"
	case DeleteBehaviourProtect:
		return "protect"
	}
	return "unknown"
}
